---
title: "IMPRINTS-CETSA-vignette"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{IMPRINTS-CETSA-vignette}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Introduction  
This package is primarily designed to process the IMPRINTS format of MS-CETSA results, first introduced in [Dai et al. 2018 Cell](https://www.cell.com/cell/fulltext/S0092-8674(18)30397-0).  

The main tasks that can be accomplished in this package include:  
1. Import the quantitative data (search result txt.file exported from Proteome Discoverer, PD) into R/Rstudio  
2. Clean up the data, mainly to remove the entries without quantitative information  
3. Rearrange data into one unique protein per row format  
4. Normalize data based on the principle of minimal variance should be controlled for different samples under each heating condition (i.e.,treatment/replicate combination)  
5. Calculate the relative fold change between treatments for each temperature/replicate combination  
6. Calculate the protein abundance score and stability score, segregate proteins into a few possible categories, such as NN, NC, CN or CC  
7. Visualize the data using a customizable bar plot layout, either for all data or for the subsets  
8. Map proteins to known complexes, calculate the correlation of IMPRINTS profiles among the protein complex subunits  
9. Extrapolate the IMPRINTS signal to a reference melting curve, for further visual and mathematical control  

--------------------------------------------------------------------------------------------

## Prerequisites
* R version >= 4.0.0, preferably the most updated version  
* Rstudio version > 1.0, preferably the most updated version  
* Imported packages:  
    arrayQualityMetrics,
    Biobase,
    dplyr (>= 1.0.0),
    drc (>= 3.0),
    fdrtool,
    GGally,
    ggplot2 (>= 3.2.0),
    ggpubr,
    ggrepel (>= 0.8.0),
    graphics,
    grDevices,
    grid,
    gridExtra,
    gridSVG,
    gtools,
    limma,
    MESS,
    methods,
    mice,
    Nozzle.R1,
    plyr (>= 1.8.0),
    RColorBrewer,
    readr (>= 2.0.0),
    reshape2 (>= 1.4.0),
    scales,
    stats,
    tibble (>= 3.0.0),
    tidyr (>= 1.2.0),
    utils,
    VennDiagram,
    VIM,
    XML
During the package installation process from local source, when these dependent packages are not available, complains would rise, use `install.packges()` or `BiocManager::install()` command to manually install the required supporting packages (as specified in the warning message), till there is no more complain.  

## Proceessing
1. For better organization of data analysis, it is __highly recommended__ to use the [Project management feature](https://support.rstudio.com/hc/en-us/articles/200526207-Using-Projects) in Rstudio.  
For each project, you are suggested to create a brand new working folder as your local working directory.  
Save the PD derived data files to this working directory folder.  
At the right up corner of Rstudio, click Project -> New Project -> Existing Directory -> browse and choose the designated working directory folder.  
By establishing a specific project for each experiment, all the data content and execution history will be saved in the project for future analysis even when R session is terminated.  

2. Activate `IMPRINTS.CETSA` package by library it.  
```{r setup}
library(IMPRINTS.CETSA)
```

3. Read in data from result files, `imprints_rawread()` is designed for read in one or many files in a vector of file names.  
This function would extract the protein abundance values and use for systematic normalization in the next step.  
Note that for simplicity, during the data search in PD software, always use channel __126__ as control channel for data search in PD, regardless of the labeling arrangement.  
You need to specify a `treatment` vector of the treatment used in the same order as the sample being labeled from 126 up to 131, ideally the replicate information should also be included, the replicate and treatment characters should be separated by an underline (_).  
Note that both replicate and treatment character should not contain neither underline nor space, in addition they should not start with number or other special character.  
The ProteinGroup data exported from Proteome Discoverer (PD) or MaxQuant is currently supported. Default is PD.  
```{r, eval=FALSE}
Chem <- imprints_rawread(c("Chem_37C_3bio_Proteins.txt",
                           "Chem_47C_3bio_Proteins.txt",
                           "Chem_50C_3bio_Proteins.txt",
                           "Chem_52C_3bio_Proteins.txt",
                           "Chem_54C_3bio_Proteins.txt",
                           "Chem_57C_3bio_Proteins.txt"),
                         treatment=c("B1_G1S","B1_S","B1_PM","B2_G1S","B2_S","B2_PM","B3_G1S","B3_S","B3_PM","Mix"))
```

4. Customize the condition names in dataset using `ms_conditionrename()`.  
When read in multiple files, a number suffix is added to the condition of each data file in sequence. This feature seems redundant but could allow for the retrieval of files read in sequence, and more importantly, make sure the condition names are always unique and distinguishable.  
In IMPRINTS-CETSA, one heating temperature is used for each condition. When multiple sets of samples are present in a complex experimental layout, the user should use a format of _"set_temperature"_.   
The argument `incondition` corresponds to the vector of current condition namings, whereas `outcondition` correponds to the vector of new condition namings, both order and length should match exactly.  
```{r, eval=FALSE}
Chem <- ms_conditionrename(Chem, incondition = c("37C.1","47C.2","50C.3","52C.4","54C.5","57C.6"),
                           outcondition=c("37C","47C","50C","52C","54C","57C"))
# to remove the "Mix" channel sample
Chem <- Chem[,-grep("^Mix$", colnames(Chem))]
```

5. Remove proteins without quantitative information by default using `ms_clean()`.  
By default the Contaminant proteins ("Cont_" in protein ID), Bos taurus proteins, Keratin family proteins, Bovine serum proteins, and Human Trypsin enzymes are removed. The default parameter for prefixcontaminant is "Cont".  
By default the orphan proteins that even only appear in one of the datasets are NOT removed, however you can specify `remsinglecondprot=TRUE`, these orphan proteins will be excluded from downstream analysis.  
```{r, eval=FALSE}
Chem_c <- ms_clean(Chem_c)
# to only keep the proteins from the right species, in most cases, this is no more necessary, 
# because this have been achieved in the ms_clean() function  
Chem_c <- Chem_c[grep("Homo sapiens", Chem$description), ]
```

6. Resolve the possible ambiguity of protein isoforms from multiple runs of searches using `ms_isoform_resolve()`.  
When two or more isoforms of the same parental proteins are present in the combined datasets but within each individual condition, only one isoform is identified, according to the principle of parsimony, the minor isoforms will be automatically adjusted to be the corresponding major or more frequently observed isoform. At the same time, a match table will be suggested to further consolidate the cases that when two or more isoforms of the same parental proteins are identified within the same individual conditioned dataset, which could then be achieved by using `ms_isoform_consolidate()`.   
```{r, eval=FALSE}
Chem_c1 <- ms_isoform_resolve(Chem_c)
# The next isoform cleaning is optional.  
# You should and are encouraged to double check and adjust the match table entries.  
Chem_c2 <- ms_isoform_consolidate(Chem_c1, nread=9, matchtable = "./subfolder/tobe_consolidated.txt")
```

7. Rearrange the dataset into the format of one unique protein per row using `imprints_rearrange()`.  
The arguments and its default value include:  
    * `nread=9`, how many readings in the dataset, default is 9 for 3X3 IMPRINTS format  
    * `repthreshold=0.75`, the minimal percentage threshold of protein being sampled from multiple runs, default value is 75%, user should adjust accordingly  
    * `averagecount=TRUE`, whether to median average the supporting PSM/peptide/count numbers, default set to TRUE  
    * `countthreshold=2`, the minimal threshold number of associated abundance count of proteins, default value is 2  
Output in working directory:  
* A copy of the rearranged dataset are written into local directory  
```{r,eval=FALSE}
Chem_c3 <- imprints_rearrange(Chem_c2, nread=9, repthreshold=0.8, countthreshold=1)
Chem_c3 <- imprints_rearrange(Chem_c2, nread=9, repthreshold=1, countthreshold=3)
```

8. Perform a systematic normalization on the whole dataset using `imprints_normalization()`.
In this function, a VSN-based model normalization is first applied, and then the function ‘removeBatchEffect’ from limma R package is called to remove potential batch effect.  
```{r,eval=FALSE}
Chem_s <- imprints_normalization(Chem_c3)
```

(Pause step) The user should check the normalization report in the local directory before proceeding to the next step.  

9. Calculate the relative fold change between treatment for each temperature/replicate combination in the dataset using `imprints_caldiff()`.  
The arguments and its default value include:  
    * `reftreatment`, the name of treatment used as the reference control to calculate the relative fold changes, such as "DMSO" or "Ctrl"  
```{r,eval=FALSE}
Chem_s1 <- imprints_caldiff(Chem_s, reftreatment="G1S")
```

10. (Optional) To retrieve a reproducible subset of the IMPRINTS-CETSA dataset using `imprints_reproducible()`, regarding to expression level changes and thermal shifts  
The arguments and its default value include:  
    * `cvthreshold` the CV threshold value for subsetting reproducible measurements, default value is 0.1  
    * `corrthreshold` the Correlation threshold value for subsetting reproducible measurements, default value is 0.5  
```{r, eval=FALSE}
Chem_s2 <- imprints_reproducible(Chem_s1)
```

11. Carry out data segregation based on the relative protein expression level change and thermal shift using `imprints_score()`, the basic definition formula was adapted from Becher et al. 2018.  
Depending on the chosen format value, the `imprints_score()` function can segregate the proteins into "NN", "CN", "NC", and "CC" four categories, or into 9 categories: "NN", "CN+", "CN-", "NC+", "NC-", "CC++", "CC+-", "CC-+", "CC--". These categories are named after the abbreviation that "C" means "Changes" while "N" means "No Changes", and the first letter indicates the change in protein expression level while the second letter indicates the change in thermal shift. The "+" or "-" sign in the 9-category format is determined by the direction of changes of expression and thermal stability, respectively, i.e., the sign of abundance.score and stability.score.mean.  

The arguments and its default value include:  
    * `format` a character choosing between 9 and 4, indicating how many categories to segregate, default value is 9  
    * `basetemp` a character indicating the baseline temperature used for protein level measurement, default value is 37C  
    * `allzscore` whether to use all the readings from all the different treatment groups when calculating the z score, default set to TRUE  
    * `fdrthreshold` the significance level of global fdr, default value is 0.01  
    * `useMAD` whether to use MAD scheme for significance analysis, default set to FALSE  
    * `nMAD` the significance level of MAD, default value 2.5  
    * `labelnodes` whether to text label the selected nodes in the graph, default set to FALSE  
    * `labelcategory` the categories of nodes to label in the graph, default is c("CC","NC","CN")  
    * `labelgeneid` a vector of the gene symbol id to show on the plot, this is exclusive from labelcategory  
```{r, eval=FALSE}
Chem_score <- imprints_score(Chem_s1, fdrthreshold=0.05, labelnodes=TRUE, labelcategory="CC+-")
Chem_score_mad <- imprints_score(Chem_s1, useMAD=TRUE, nMAD=2.5, labelnodes=TRUE, labelcategory="CC+-")
Chem_score_4 <- imprints_score(Chem_s1, format="4", labelnodes=TRUE, labelcategory="CC")

# could use the reproducible subset if the previous step was performed
Chem_score_r <- imprints_score(Chem_s2, labelnodes=TRUE, labelcategory="CC+-")
Chem_score_4r <- imprints_score(Chem_s2, format="4", labelnodes=TRUE, labelcategory="CC")
```

12. Plot out the bar graph plot for each protein using `imprints_barplotting()`.  
The arguments and its default value includes:  
    * `treatmentlevel`, a vector to specify the treatment keyword information, the order determines the color arrangement as set in colorpanel  
    * `setlevel`, a vector to specify the set key word information if any  
    * `colorpanel=c("gray", "blue", "orange")` a vector to customize the colors. The user is encouraged to refer to [Colors in R](http://www.stat.columbia.edu/~tzheng/files/Rcolor.pdf) or [R Color Cheatsheet](https://www.nceas.ucsb.edu/~frazier/RSpatialGuides/colorPaletteCheatsheet.pdf) for more color selection.  
```{r, eval=FALSE}
# plot out some proteins   
cyclin <- Chem_s1[grep("[Cc]yclin[- ]",Chem_s1$description),]
imprints_barplotting(cyclin, treatmentlevel=c("G1S","S","PM"))
# plot out the proteins from one category  
Chem_CC <- subset(Chem_s1, id %in% subset(Chem_score_4, category=="CC")$id)
imprints_barplotting(Chem_CC, treatmentlevel=c("G1S","S","PM"))
# plot out all the proteins  
imprints_barplotting(Chem_s1, treatmentlevel=c("G1S","S","PM"))
```

13. Map proteins onto known complexes, calculate the correlation of IMPRINTS profiles among the complex subunits using `imprints_complex_mapping()` and `imprints_corr_in_complex()`.  
The input dataset should be the average IMPRINTS profile, i.e., after `imprints_average()`, make sure the columns with readings are named in the format like "37C_Treatment". In addition, the category information in the dataset from `imprints_score()` should be provided, to allow focus on a subset of the data, say the "Changed" proteins.  
The arguments and its default value in `imprints_complex_mapping()` includes:  
    * `categorytable` dataset after `imprints_score()`  
    * `set` a single character to specify the sample name to analyze if any  
    * `treatment` a single character to specify the sample name to analyze  
    * `targetcategory=c("NC","CN","CC")` by default only to use the proteins with either abundance or stability changes  
    * `complexdatabase="Corum"` the default complex database used is core set of CORUM database, can use `View(coreCorum)` to check the available complexes   
    * `organism="Human"` user can switch to Mouse or Rat if necessary  
    * `complexID` if provided, only the specified complexes were used, default to NULL, ie, use all the possible complexes from the whole database  
    * `minsubunitsIdentified` the minimal number of subunits in the data, default value is 3  

The arguments and its default value in `imprints_corr_in_complex()` includes:  
    * `nread=6`, how many readings in the dataset, default is 6 temperatures  
    * `removeredundancy` scrutinize the complex input, to remove the redundancy according to the measured subunits, default set to FALSE  
    * `similaritythreshold` the threshold for similarity, useful only when `removeredundancy=TRUE`, default value 1.0 to only remove the complete overlap ones  
```{r, eval=FALSE}
# to first average the data
Chem_s1_ave <- imprints_average(Chem_s1)
# to check the proteins in PM with Changes to complex
Chem_PM_complex <- imprints_complex_mapping(Chem_s1_ave, Chem_score_4,
                                          treatment="PM", targetcategory=c("NC","CN","CC"), 
                                          organism="Human")
# to calculate the correlation of IMPRINTS profiles 
Chem_PM_complex_corr <- imprints_corr_in_complex(Chem_PM_complex, nread=6)
# to remove the redundant complexes with highly similar composition of protein subunits
Chem_PM_complex_corr <- imprints_corr_in_complex(Chem_PM_complex, nread=6, 
                                                 removeredundancy=T)
```

14. Calculate the correlation of the IMPRINTS profiles for the same protein in two different treatments using `imprints_correlation()`.  
The arguments and its default value in `imprints_correlation()` includes:  
    * `data` the input data is typically after `imprints_average()`  
    * `setvector` a two-element character vector with distinguishable keyword from the column names of the input dataset if any  
    * `treatmentvector` a two-element character vector with distinguishable keyword from the column names of the input dataset  

```{r, eval=FALSE}
# It is possible to look at the whole dataset, but it is not recommended to run like below
# Chem_s1_corr <- imprints_correlation(Chem_s1_ave, treatmentvector=c("S","PM"))
# It is probably more useful to only look at the "Changed" proteins
Chem_s1_ave_C <- subset(Chem_s1_ave, id %in% subset(Chem_score_4,      
                                                    category%in%c("CC","NC","CN"))$id)
Chem_s1_corr1 <- imprints_correlation(Chem_s1_ave_C, treatmentvector=c("S","PM"))

# Then it is possible to plot these "Changed" proteins in a descending order of the correlations, which is controlled by specifying the correlation information  
Chem_s1_C <- subset(Chem_s1, id %in% subset(Chem_score_4, category%in%c("CC","NC","CN"))$id)
imprints_barplotting(Chem_s1_C, treatmentlevel=c("G1S","S","PM"), corrtable=Chem_s1_corr1)
```

15. It is also possible to do standard protein level differential expression analysis from IMPRINTS dataset, using `imprints_diffExp()`.  
    * `data` the input dataset after `imprints_normalization()` function, readings in log2 format  
    * `set` a single character to specify the sample name if any  
    * `contrast` a character to indicate the contrasting treatment conditions, typically in a "Condition1-Condition2" format   
    * `logFC_threshold` the threshold value for log fold changes, default set at 0.3  
    * `adjp_threshold` the threshold value for adjusted p values, default set at 0.01  
```{r, eval=FALSE}
Chem_PM_de <- imprints_diffExp(Chem_s, contrast="PM-G1S")
Chem_S_de <- imprints_diffExp(Chem_s, contrast="S-G1S")
```
